daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_functions/BitFunctions.tdml

```diff
@@ -0,0 +1,112 @@
+        </xs:element>
+        <xs:element name="rightInt">
+          <xs:complexType>
+            <xs:sequence dfdl:separator=",">
+              <xs:element name="num" type="xs:int" dfdl:lengthKind="delimited" dfdl:textNumberPattern="#0"/>
+              <xs:element name="count" type="xs:int" dfdl:lengthKind="delimited"/>
+              <xs:element name="value" type="xs:int" dfdl:inputValueCalc="{dfdlx:rightShift(../num,../count)}"/>
+            </xs:sequence>
+          </xs:complexType>
+        </xs:element>
+  </tdml:defineSchema>
+  <tdml:parserTestCase name="leftShift_int_01" root="leftInt" model="BitFunctions">
+              <tdml:document>22,2</tdml:document>
+              <tdml:infoset>
+                <tdml:dfdlInfoset>
+                    <ex:leftInt>
+                        <num>22</num>
+                        <count>2</count>
+                        <value>88</value>
+                    </ex:leftInt>
+                </tdml:dfdlInfoset>
+              </tdml:infoset>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="leftShift_int_02" root="leftInt" model="BitFunctions">
+              <tdml:document>2.2,2</tdml:document>
+                <tdml:errors>
+                  <tdml:error>Unable to parse xs:int from text: 2.2</tdml:error>
+                </tdml:errors>
+  </tdml:parserTestCase>
+
+    <tdml:parserTestCase name="leftShift_int_03" root="leftInt" model="BitFunctions">
+                <tdml:document>100000,10</tdml:document>
+                <tdml:infoset>
+                  <tdml:dfdlInfoset>
+                      <ex:leftInt>
+                          <num>100000</num>
+                          <count>10</count>
+                          <value>102400000</value>
+                      </ex:leftInt>
+                  </tdml:dfdlInfoset>
+                </tdml:infoset>
+    </tdml:parserTestCase>
+
+        <tdml:parserTestCase name="leftShift_int_04" root="leftInt" model="BitFunctions">
+                    <tdml:document>100000,25</tdml:document>
+                    <tdml:infoset>
+                      <tdml:dfdlInfoset>
+                          <ex:leftInt>
+                              <num>100000</num>
+                              <count>25</count>
+                              <value>1073741824</value>
+                          </ex:leftInt>
+                      </tdml:dfdlInfoset>
+                    </tdml:infoset>
+        </tdml:parserTestCase>
+
+
+    <tdml:parserTestCase name="rightShift_int_01" root="rightInt" model="BitFunctions">
+                <tdml:document>4,2</tdml:document>
+                <tdml:infoset>
+                  <tdml:dfdlInfoset>
+                      <ex:rightInt>
+                          <num>4</num>
+                          <count>2</count>
+                          <value>1</value>
+                      </ex:rightInt>
+                  </tdml:dfdlInfoset>
+                </tdml:infoset>
+    </tdml:parserTestCase>
+
```

Comment:
Ok, so we need to add more tests, to evoke the errors if the shifts are too big, to cover the case of shifting by exactly the width (e.g., 32 for an int), and shifting -1 values right should sign-extend for right-shift of signed types. Also shift count of 0 should be allowed for both right and left for all types. 

You may not want to create individual test cases for these. You can create a single larger test with a bunch of elements in it, that tests several things simultaneously.

Reply:
If I create multiple infoset in one test It just takes the 1st infoset and ignores the rest.
How can I create a test case with bunch of elements?

Reply:
Ok, start by adding a left-shift and right-shift complex type to your defineSchema
E.g.,, 
`
          <xs:complexType name="leftIntType">
            <xs:sequence dfdl:separator=",">
              <xs:element name="num" type="xs:int" dfdl:lengthKind="delimited" dfdl:textNumberPattern="#0"/>
              <xs:element name="count" type="xs:int" dfdl:lengthKind="delimited"/>
              <xs:element name="value" type="xs:int" dfdl:inputValueCalc="{dfdlx:leftShift(../num,../count)}"/>
            </xs:sequence>
          </xs:complexType>

          <xs:complexType name="rightInt">
            <xs:sequence dfdl:separator=",">
              <xs:element name="num" type="xs:int" dfdl:lengthKind="delimited" dfdl:textNumberPattern="#0"/>
              <xs:element name="count" type="xs:int" dfdl:lengthKind="delimited"/>
              <xs:element name="value" type="xs:int" dfdl:inputValueCalc="{dfdlx:rightShift(../num,../count)}"/>
            </xs:sequence>
          </xs:complexType>
`
Now you can create an element declaration that has many leftInt and rightInt elements in it.
`
<xs:element name="testIntShifting">
   <xs:complexType>
     <xs:sequence dfdl:separator="|"> 
      <xs:element name="leftInt1" type="ex:leftIntType"/>
      <xs:element name="leftInt2" type="ex:leftintType"/>
      <!-- as many as you want here -->
      <xs:element name="rightInt1" type="ex:rightIntType"/>
      <xs:element name="rightInt2" type="ex:rightIntType"/>
      <!-- as many as you want here -->
   </xs:sequence>
</xs:complexType>
</xs:element> 
`
Now you create a parser test case that uses this new more complicated element. For example:
`
    <tdml:parserTestCase name="intShiftTest1" root="testIntShifting" model="BitFunctions">
                <tdml:document>1,2|1,0|4,2|4,0</tdml:document>
                <tdml:infoset>
                  <tdml:dfdlInfoset>
                      <ex:testIntShifting>
                      <leftInt1>
                          <num>1</num>
                          <count>2</count>
                          <value>4</value>
                      </leftInt1>
                      <leftInt2>
                         <num>1</num>
                        <count>0</count>
                       <value>1</value>
                     </leftInt2>
                     <rightInt1>
                        <num>4</num>
                       <count>2</count>
                       <value>1</value>
                    </rightInt1>
                    <rightInt2>
                       <num>4</num>
                       <count>0</count>
                       <value>4</value>
                    </rightInt2>
                </ex:testIntShifting>
                  </tdml:dfdlInfoset>
                </tdml:infoset>
    </tdml:parserTestCase>
`
That lets you test 4 variations at a time.

Reply:
As a small tweak, you could also do something like this:

```xml
<xs:element name="testIntLeftShifting">
   <xs:complexType>
     <xs:sequence dfdl:separator="|"> 
      <xs:element name="leftInt" type="ex:leftIntType" maxOccurs="unbonded" />
   </xs:sequence>
  </xs:complexType>
</xs:element>
```
Then you can have as many value/shift pairs as you, which each one separated by a pipe, and the infoset will just have a bunch of ``<leftInt>`` elements for each pair.